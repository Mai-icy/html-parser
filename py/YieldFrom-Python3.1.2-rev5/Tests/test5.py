#
#   Test raising exception in delegated next() call
#

def g1():
	try:
		print("Starting g1")
		yield "g1 ham"
		yield from g2()
		yield "g1 eggs"
	finally:
		print("Finishing g1")

def g2():
	try:
		print("Starting g2")
		yield "g2 spam"
		raise ValueError("hovercraft is full of eels")
		yield "g2 more spam"
	finally:
		print("Finishing g2")

for x in g1():
	print("Yielded", x)
