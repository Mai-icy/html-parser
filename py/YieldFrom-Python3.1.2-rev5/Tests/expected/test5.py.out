Starting g1
Yielded g1 ham
Starting g2
Yielded g2 spam
Finishing g2
Finishing g1
Traceback (most recent call last):
  File "test5.py", line 23, in <module>
    for x in g1():
  File "test5.py", line 9, in g1
    yield from g2()
  File "test5.py", line 18, in g2
    raise ValueError("hovercraft is full of eels")
ValueError: hovercraft is full of eels
